src/api/spec/controllers/webui/project_controller_spec.rb

```diff
@@ -914,6 +914,49 @@
     end
   end
 
+  describe 'POST #save_prjconf' do
+    before do
+      login user
+    end
+
+    context 'can save a project config' do
+      before do
+        post :save_prjconf, params: { project: user.home_project.name, config: 'save config' }
+      end
+
+      it { expect(flash[:success]).to eq('Config successfully saved!') }
+      it { expect(response.status).to eq(200) }
+    end
+
+    context 'cannot save a project config' do
+      before do
+        allow_any_instance_of(ProjectConfigFile).to receive(:save).and_return(nil)
+        post :save_prjconf, params: { project: user.home_project.name, config: '' }
+      end
+
+      it { expect(flash[:error]).not_to be_nil }
+      it { expect(response.status).to eq(400) }
+    end
+
+    context 'cannot save with an unauthorized user' do
+      before do
+        post :save_prjconf, params: { project: another_project.name, config: 'save config' }
+      end
+
+      it { expect(flash[:error]).to eq('Sorry, you are not authorized to update this Project.') }
+      it { expect(response.status).to eq(302) }
+      it { expect(response).to redirect_to(root_path) }
+    end
+
+    context 'with an non existing project' do
+      let(:post_save_prjconf) { post :save_prjconf, params: { project: 'non:existing:project', config: 'save config' } }
+
+      it 'raise a RecordNotFound Exception' do
+        expect{ post_save_prjconf }.to raise_error ActiveRecord::RecordNotFound
```

Comment:
To `raise_error` the `expect` needs a block to catch the exception, because the block isn't executed immediately.